Code/Framework/AzCore/AzCore/RTTI/AzStdOnDemandReflectionSpecializations.cpp

```diff
@@ -9,18 +9,19 @@
 #include <AzCore/RTTI/BehaviorContext.h>
 #include <AzCore/RTTI/ReflectContext.h>
 #include <AzCore/RTTI/TypeInfo.h>
-#include <AzCore/std/string/alphanum.h>
 #include <AzCore/std/string/conversions.h>
 #include <AzCore/std/string/string.h>
-#include <AzCore/std/string/tokenize.h>
+#include <AzCore/StringFunc/StringFunc.h>
 
 #include <AzCore/RTTI/AzStdOnDemandReflection.inl>
 #include <AzCore/RTTI/AzStdOnDemandReflectionLuaFunctions.inl>
+
 namespace AZ::CommonOnDemandReflections
 {
-    void ReflectCommonString(ReflectContext* context)
+    template<class T>
+    void ReflectCommonStringAPI(ReflectContext* context)
```

Comment:
I only die a little inside when I see these classes reflected in lua like basic_string_char_traits_char_ref, I'm guessing if we provided some friendlier class names like "String" it would break existing code?

Reply:
We do provide friendlier names for [ScriptCanvas](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/RTTI/AzStdOnDemandPrettyName.inl#L53-L60) at least